src/renderers/react/externalStore.ts

```diff
@@ -0,0 +1,70 @@
+import type { ReactElement } from "react";
+import { useEffect, useState } from "react";
+
+/**
+ * Creates a store to allow state within React to be programmatically updated
+ * outside of a React context.
+ */
+export const createStore = <Value>(getValue: Value) => {
+  type Subscription = (val: Value) => void;
+
+  const subscribers: Subscription[] = [];
+
+  const store = {
+    /**
+     * Update the value in the store, informing subscribers.
+     *
+     * @return {boolean} true if there are any subscribers, false if not.
+     */
+    update: (val: Value): boolean => {
+      subscribers.forEach((sub) => sub(val));
+      return subscribers.length ? true : false;
```

Comment:
Out of interest, what's the rationale / value behind this returning a boolean? Would a number of subscribers be as useful?

Reply:
We could also do that! It's to ensure that we're able to test unsubscription without having to dig in to implementation detail of the store.